srml/support/procedural/src/storage/transformation.rs

```diff
@@ -234,9 +245,10 @@ fn decl_store_extra_genesis(
 					)
 				);
 			};
-			if type_infos.kind.is_simple() && ext::has_parametric_type(type_infos.value_type, traitinstance) {
+
+			if type_infos.kind.is_simple()
+				&& ext::type_contains_ident(type_infos.value_type, traitinstance) {
```

Comment:
this was already wrong but we have to set is_trait_needed for other kind as well (see test attached)

[t.txt](https://github.com/paritytech/substrate/files/3313955/t.txt)